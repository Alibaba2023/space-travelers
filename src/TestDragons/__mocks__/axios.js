const dragonsData = [
  {
    id: 'dragon1',
    name: 'Dragon 1',
    description: 'Dragon 1 description',
    flickr_images: ['image1.jpg'],
  },
  {
    id: 'dragon2',
    name: 'Dragon 2',
    description: 'Dragon 2 description',
    flickr_images: ['image2.jpg'],
  },
];

const rocketsData = [
  {
    id: 'rocket1',
    rocketName: 'Rocket 1',
    description: 'Rocket 1 description',
    imageRocket: 'rocket1.jpg',
  },
  {
    id: 'rocket2',
    rocketName: 'Rocket 2',
    description: 'Rocket 2 description',
    imageRocket: 'rocket2.jpg',
  },
];

const get = (url) => {
  if (url === 'https://api.spacexdata.com/v3/dragons') {
    return Promise.resolve({
      data: dragonsData,
    });
  }

  if (url === 'https://api.spacexdata.com/v4/rockets') {
    return Promise.resolve({
      data: rocketsData,
    });
  }

  return Promise.resolve({
    data: {},
  });
};

export default { get };
