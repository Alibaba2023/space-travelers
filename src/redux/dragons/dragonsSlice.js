import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  dragons: [],
  isLoading: true,
  error: null,
};

const dragonsSlice = createSlice({
  name: 'dragons',
  initialState,
  reducers: {

  },
});

export const { dragonsLoading, dragonsReceived } = dragonsSlice.actions;
export default dragonsSlice.reducer;
