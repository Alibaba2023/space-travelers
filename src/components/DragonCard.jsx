import React from 'react';
import PropTypes from 'prop-types';
import '../style/dragons.css';

const DragonCard = ({ dragon }) => (
  <div className="dragon-card rounded my-3" key={dragon.id}>
    <div className="row">
      <div className="col-sm-12 col-md-5">
        {dragon.name === 'Dragon 1' ? (
          <img className="w-100" src={dragon.flickr_images[1]} alt={dragon.name} />
        ) : (
          <img className="w-100" src={dragon.flickr_images[2]} alt={dragon.name} />
        )}
      </div>
      <div className="col-sm-12 col-md-7">
        <div className="py-3">
          <h2 className="">
            {dragon.name}
          </h2>
        </div>
        <div className="">{dragon.description}</div>
        <div className="py-3">
          <button type="submit" className="btn btn-primary">Reserve Dragon</button>
        </div>
      </div>
    </div>
  </div>
);

DragonCard.propTypes = {
  dragon: PropTypes.shape({
    id: PropTypes.string.isRequired,
    flickr_images: PropTypes.arrayOf(PropTypes.string).isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired,
};

export default DragonCard;
