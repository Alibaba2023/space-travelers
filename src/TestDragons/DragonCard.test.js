import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import axios from './__mocks__/axios';
import DragonCard from '../components/DragonCard';
import dragonsSlice from '../redux/dragons/dragonsSlice';

jest.mock('axios');

const store = configureStore({
  reducer: {
    dragons: dragonsSlice,
  },
});

test('renders a dragon card correctly', async () => {
  // Define los datos del dragón a ser mostrados en el componente DragonCard
  const dragon = {
    id: 'dragon1',
    name: 'Dragon 1',
    description: 'Dragon 1 description',
  };

  // Mockea la respuesta de axios cuando la URL sea "https://api.spacexdata.com/v3/dragons"
  axios.get.mockResolvedValueOnce({
    data: axios.dragonsData, // dragonsData es la información que retornamos en el archivo axios.js
  });

  const { container } = render(
    <Provider store={store}>
      <DragonCard dragon={dragon} />
    </Provider>,
  );

  expect(container.firstChild).toMatchSnapshot();
  expect(screen.getByText('Dragon 1')).toBeInTheDocument();
  expect(screen.getByText('Dragon 1 description')).toBeInTheDocument();
  // ... Añade más expectativas según la lógica de tu componente DragonCard
});
