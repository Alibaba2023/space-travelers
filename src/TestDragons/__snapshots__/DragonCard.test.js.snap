// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a dragon card correctly 1`] = `
<div
  class="dragon-card rounded my-3"
>
  <div
    class="row"
  >
    <div
      class="col-md-5 col-sm-12 d-flex justify-content-around"
    >
      <div>
        No image available
      </div>
    </div>
    <div
      class="col-sm-12 col-md-7"
    >
      <div
        class="py-3"
      >
        <h2
          class=""
        >
          Dragon 1
        </h2>
      </div>
      <div
        class=""
      >
        Dragon 1 description
      </div>
      <div
        class="py-3"
      >
        <button
          class="btn btn-primary"
          type="submit"
        >
          Reserve Dragon
        </button>
      </div>
    </div>
  </div>
</div>
`;
